spock-specs/src/test/groovy/org/spockframework/geantyref/GenericTypeReflectorUtilSpec.groovy

```diff
@@ -0,0 +1,135 @@
+
+  def "erase"(Type inputType, Class<?> expectedClass) {
+    expect:
+    GenericTypeReflectorUtil.erase(inputType) == expectedClass
+
+    where:
+    inputType             | expectedClass
+    Object                | Object
+    StringBuilder         | StringBuilder
+    FUNC_STRING_INT_TYPE  | Function
+    LIST_STRING_TYPE      | List
+    LIST_LIST_STRING_TYPE | List
+  }
+
+  def "getTypeName"(Type inputType, String expectedName) {
+    expect:
+    GenericTypeReflectorUtil.getTypeName(inputType) == expectedName
+
+    where:
+    inputType             | expectedName
+    Object                | "java.lang.Object"
+    StringBuilder         | "java.lang.StringBuilder"
+    FUNC_STRING_INT_TYPE  | "java.util.function.Function<java.lang.String, java.lang.Integer>"
+    LIST_STRING_TYPE      | "java.util.List<java.lang.String>"
+    LIST_LIST_STRING_TYPE | "java.util.List<java.util.List<java.lang.String>>"
+  }
+
+  def "getParameterTypes"(TestMethod testMethod, List<Type> expectedTypes) {
+    expect:
+    GenericTypeReflectorUtil.getParameterTypes(testMethod.method, testMethod.declaringClass).toList() == expectedTypes
+
+    where:
+    testMethod                               | expectedTypes
+    method(Object, "equals")                 | [Object]
+    method(List, "equals")                   | [Object]
+    method(NodeInfo, "getAnnotationsByType") | [Class]
+  }
+
+  private abstract class TestNodeInfo extends NodeInfo<TestNodeInfo, Method> {}
+
+  StringBuilder exactReturnType() { null }
+
+  public <T> T genericMethod(Class<T> arg) { null }
+
+  public <T extends Serializable> T genericMethodWithTypeBound(Class<T> arg) { null }
+
+  public <T extends Serializable & Runnable> T genericMethodWithTwoTypeBounds(Class<T> arg) { null }
+
+  def "getReturnType"(TestMethod testMethod, Type expectedType) {
+    expect:
+    GenericTypeReflectorUtil.getReturnType(testMethod.method, testMethod.declaringClass) == expectedType
+
+    where:
+    testMethod                               | expectedType
+    method(Object, "equals")                 | boolean
+    method(List, "equals")                   | boolean
+    method("exactReturnType")                | StringBuilder
+    method("genericMethod")                  | Object
+    method("genericMethodWithTypeBound")     | Serializable
+    method("genericMethodWithTwoTypeBounds") | Object
+    method(NodeInfo, "getAnnotation")        | Annotation
+    method(NodeInfo, "getName")              | String
+    method(NodeInfo, "getReflection")        | AnnotatedElement
+    method(TestNodeInfo, "getReflection")    | Method
+    method(NodeInfo, "getParent")            | NodeInfo
+    method(TestNodeInfo, "getParent")        | TestNodeInfo
+    method(ArrayList, "get")                 | Object
+  }
+
+  private static TestMethod method(Class<?> declaringClass = GenericTypeReflectorUtilSpec.class, String methodName) {
+    def method = declaringClass.getMethods().find { it.name == methodName }
```

Comment:
We should be able to use property-style access instead of getters
```suggestion
    def method = declaringClass.methods.find { it.name == methodName }
```